File: Semester 4/Medii de proiectare si programare/Laborator/Laborator 4/FestivalMuzicaJava/src/main/java/domain/Ticket.java
package domain;

import lombok.Getter;
import lombok.Setter;

import java.util.Objects;

@Getter
@Setter
public class Ticket implements Entity<Long> {
    private Long id;
    private String clientName;
    private Show show;
    private int noSeats;

    public Ticket(Long id, String clientName, Show show, int noSeats) {
        this.id = id;
        this.clientName = clientName;
        this.show = show;
        this.noSeats = noSeats;
    }

    @Override
    public Long getId() {
        return id;
    }

    @Override
    public void setId(Long id) {
        this.id = id;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Ticket ticket = (Ticket) o;
        return noSeats == ticket.noSeats && Objects.equals(id, ticket.id) && Objects.equals(clientName, ticket.clientName) && Objects.equals(show, ticket.show);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id, clientName, show, noSeats);
    }

    @Override
    public String toString() {
        return "Ticket{" +
                "id=" + id +
                ", clientName='" + clientName + '\'' +
                ", show=" + show +
                ", noSeats=" + noSeats +
                '}';
    }
}

File: Semester 4/Medii de proiectare si programare/Laborator/Laborator 5/FestivalMuzicaCSharpWFA/FestivalMuzicaCSharpWFA/Domain/Ticket.cs
namespace FestivalMuzicaCSharp.Domain;

public class Ticket : IEntity<long>
{
    public long Id { get; set; }
    public string ClientName { get; set; }
    public Show Show { get; set; }
    public int NoSeats { get; set; }
    
    public Ticket(long id, string clientName, Show show, int noSeats)
    {
        Id = id;
        ClientName = clientName;
        Show = show;
        NoSeats = noSeats;
    }
    
    public override bool Equals(object? obj) => Equals(obj as Ticket);
    
    public bool Equals(Ticket? other)
    {
        if (other is null) return false;
        if (ReferenceEquals(this, other)) return true;

        return Id == other.Id && string.Equals(ClientName, other.ClientName, StringComparison.Ordinal)
                              && Show.Equals(other.Show)
                              && NoSeats == other.NoSeats;
    }
    
    public override int GetHashCode() => HashCode.Combine(Id, ClientName, Show, NoSeats);
    
    public override string ToString()
    {
        return $"Ticket{{id={Id}, clientName='{ClientName}', show={Show}, noSeats={NoSeats}}}";
    }
}